src/response/mod.rs

```diff
@@ -41,49 +41,60 @@ impl Response {
     /// Create a new instance from a reader.
     pub fn with_reader<R>(status: u16, reader: R) -> Self
     where
-        R: BufRead + Unpin + Send + 'static,
+        R: BufRead + Unpin + Send + Sync + 'static,
     {
-        let status = http::StatusCode::from_u16(status).expect("invalid status code");
-        let res = http::Response::builder()
-            .status(status)
-            .body(Box::pin(reader).into())
-            .unwrap();
+        let status = http_types::StatusCode::try_from(status).expect("invalid status code");
+        let mut res = http_types::Response::new(status);
+        res.set_body(Body::from_reader(reader, None));
+
         Self {
             res,
             cookie_events: vec![],
         }
     }
 
     /// Returns the statuscode.
-    pub fn status(&self) -> http::StatusCode {
+    pub fn status(&self) -> http_types::StatusCode {
         self.res.status()
     }
 
     /// Set the statuscode.
-    pub fn set_status(mut self, status: http::StatusCode) -> Self {
-        *self.res.status_mut() = status;
+    pub fn set_status(mut self, status: http_types::StatusCode) -> Self {
+        self.res.set_status(status);
         self
     }
 
     /// Insert an HTTP header.
-    pub fn set_header(mut self, key: &'static str, value: impl AsRef<str>) -> Self {
+    pub fn set_header(
+        mut self,
+        key: http_types::headers::HeaderName,
+        value: impl AsRef<str>,
```

Comment:
Ideally this could forward https://docs.rs/http-types/1.0.1/http_types/struct.Response.html?search=#method.insert_header so that we can get rid of all the `parse().unwrap()` calls.

```suggestion
        key: http_types::headers::HeaderName,
        values: impl http_types::ToHeaderValues
```

Reply:
almost makes things better:

```
the trait `http_types::headers::to_header_values::ToHeaderValues` is not implemented for `&std::string::String`
```